src/axom/inlet/SchemaCreator.hpp

```diff
@@ -130,8 +130,162 @@ class SchemaCreator
+
+  /*!
+   *****************************************************************************
+   * \return An unordered map from Table names to the child Table pointers for 
+   * this Table.
+   *****************************************************************************
+   */
+  virtual std::unordered_map<std::string, std::shared_ptr<Table>> getChildTables() = 0;
+
+  /*!
+   *****************************************************************************
+   * \brief Retrieves the matching Table.
+   * 
+   * \param [in] The string indicating the target name of the Table to be searched for.
+   * 
+   * \return The Table matching the target name. If no such Table is found,
+   * a nullptr is returned.
+   *****************************************************************************
+   */
+  virtual std::shared_ptr<Table> getTable(const std::string& tableName) = 0;
+
+  /*!
+   *****************************************************************************
+   * \brief Retrieves the matching Field.
+   * 
+   * \param [in] The string indicating the target name of the Field to be searched for.
+   * 
+   * \return The Field matching the target name. If no such Field is found,
+   * a nullptr is returned.
+   *****************************************************************************
+   */
+  virtual std::shared_ptr<Field> getField(const std::string& fieldName) = 0;
+
+  /*!
+   *****************************************************************************
+   * \brief Add an array of Boolean Fields to the input deck schema.
+   *
+   * \param [in] name Name of the array
+   * \param [in] description Description of the Field
+   *
+   * \return Shared pointer to the created Field
+   *****************************************************************************
+   */
+  virtual std::shared_ptr<Table> addBoolArray(const std::string& name,
+                                      const std::string& description = "") = 0;
+
+  /*!
+   *****************************************************************************
+   * \brief Add an array of Integer Fields to the input deck schema.
+   *
+   * \param [in] name Name of the array
+   * \param [in] description Description of the Field
+   *
+   * \return Shared pointer to the created Field
+   *****************************************************************************
+   */
+  virtual std::shared_ptr<Table> addIntArray(const std::string& name,
+                                     const std::string& description = "") = 0;
+
+  /*!
+   *****************************************************************************
+   * \brief Add an array of Double Fields to the input deck schema.
+   *
+   * \param [in] name Name of the array
+   * \param [in] description Description of the Field
+   *
+   * \return Shared pointer to the created Field
+   *****************************************************************************
+   */
+  virtual std::shared_ptr<Table> addDoubleArray(const std::string& name,
```

Comment:
Have you considered using `templates` for the add & get methods here? Would that make more sense? The add/get methods seem to have the same signature, so I am not entirely sure what is the difference in the implementation -- Perhaps, something to think about as your design evolves.

Reply:
The idea for not templating the add functions was because we lock down what types that Inlet supports (bool, int, double, and string) for correctness. I do go back and forth on this though...

Reply:
You can also lock down on the types using `AXOM_STATIC_ASSERT`....